File: problems/字符串/520.检测大写字母.cpp
#include <iostream>
#include <string>
#include <cctype>
using namespace std;

bool detectCapitalUse(string word) {
    int n = word.size();
    bool firstUpper = isupper(word[0]);
    bool allUpper = true, allLower = true;
    for (int i = 1; i < n; i++) {
        if (isupper(word[i])) allLower = false;
        else allUpper = false;
    }
    return allUpper || allLower || (firstUpper && allLower);
}

int main() {
    // 测试用例
    cout << boolalpha << detectCapitalUse("USA") << endl;      // 预期: true
    cout << detectCapitalUse("FlaG") << endl;                    // 预期: false
    cout << detectCapitalUse("Google") << endl;                  // 预期: true
    cout << detectCapitalUse("leetcode") << endl;                // 预期: true
    return 0;
}

File: problems/字符串/520.检测大写字母.py
#!/usr/bin/env python3
"""
520. 检测大写字母
"""


def detect_capital_use(word: str) -> bool:
    return word.isupper() or word.islower() or (word[0].isupper() and word[1:].islower())


if __name__ == "__main__":
    # 测试用例
    print(detect_capital_use("USA"))      # 预期: True
    print(detect_capital_use("FlaG"))     # 预期: False
    print(detect_capital_use("Google"))   # 预期: True
    print(detect_capital_use("leetcode")) # 预期: True
